aslprep/workflows/asl/hmc.py

```diff
@@ -119,6 +113,103 @@ def init_asl_hmc_wf(
         name='outputnode',
     )
 
+    if use_ge:
+        # Skip HMC, just mock up movpar and xform files with identity transforms
+        workflow.__desc__ = """\
+Head-motion correction was not performed on the ASL data.
+Instead, a temporary reference image was generated by averaging the highest contrast volumes
+within the ASL series.
+This was then registered to the ASL reference image.
+This single transform was then duplicated for each volume in the ASL series.
+
+"""
+        select_highest_contrast_volumes = pe.Node(
+            SelectHighestContrastVolumes(prioritize_m0=use_ge),
+            name='select_highest_contrast_volumes',
+            mem_gb=1,
+        )
+        workflow.connect([
+            (inputnode, select_highest_contrast_volumes, [
+                ('asl_file', 'asl_file'),
+                ('aslcontext', 'aslcontext'),
+            ]),
+        ])  # fmt:skip
+
+        gen_avg = pe.Node(RobustAverage(), name='gen_avg', mem_gb=1)
+        workflow.connect([
+            (select_highest_contrast_volumes, gen_avg, [('selected_volumes_file', 'in_file')]),
+        ])  # fmt:skip
+
+        reference_buffer = pe.Node(
+            niu.IdentityInterface(fields=['aslref']),
+            name='reference_buffer',
+        )
+
+        if config.workflow.smooth_kernel > 0:
+            # Smooth the reference image to match the aslref processing
+            workflow.__desc__ += (
+                'The reference image was then smoothed with a Gaussian kernel '
+                f'(FWHM = {config.workflow.smooth_kernel} mm).'
+            )
+            smooth_reference = pe.Node(
+                Smooth(fwhm=config.workflow.smooth_kernel),
+                name='smooth_reference',
+                mem_gb=config.DEFAULT_MEMORY_MIN_GB,
+            )
+            workflow.connect([
+                (gen_avg, smooth_reference, [('out_file', 'in_file')]),
+                (smooth_reference, reference_buffer, [('out_file', 'aslref')]),
+            ])  # fmt:skip
+        else:
+            workflow.connect([(gen_avg, reference_buffer, [('out_file', 'aslref')])])
+
+        # Now register the temporary reference image to the ASL reference image
+        register_reference = pe.Node(
+            Registration(
+                transforms=['Rigid'],
+                metric=['Mattes'],
+                metric_weight=[1],
+                shrink_factors=[[4]],
+                smoothing_sigmas=[[0]],
+                number_of_iterations=[[10000]],
```

Comment:
Probably want multiple stages here.

Reply:
```suggestion
                shrink_factors=[[4, 2, 1]],
                smoothing_sigmas=[[2, 1, 0]],
                number_of_iterations=[[10000, 10000, 10000]],
```

Downsamples by 4, then 2, then 1.